### PR TITLE
Add tests and implementation for disabling dropout layers in models

### PR DESCRIPTION
**Pull request by Ankur-singh**
#### Context
What is the purpose of this PR? Is it to
- [x] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses. 
#2353

#### Changelog
What are the changes made in this PR?
* Added `disable_dropout` function to `torchtune/training/model_util.py`
* Added tests for `disable_dropout` function

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [X] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [X] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [X] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [X] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

File: torchtune/training/model_util.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+def disable_dropout(model: torch.nn.Module) -> None:
+    """
+    Disables dropout layers in the given model.
+
+    Args:
+        model (torch.nn.Module): The model in which dropout layers should be disabled.
+    """
+    for module in model.modules():
+        if isinstance(module, torch.nn.Dropout):
```
Comment (joecummings) [inline torchtune/training/model_util.py:19]: I would also check that the Dropout set to something other than zero. No need to reset Dropout to zero if it's already zero.

File: torchtune/training/model_util.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            warnings.warn(
+                f"Dropout found in {module}. This is likely to cause issues during training. Disabling."
```
Comment (joecummings) [inline torchtune/training/model_util.py:21]:
```suggestion
                f"Found Dropout with value {module.p} in module {module}. Setting to zero."
```

**Review by joecummings — COMMENTED** [review]
High quality work!

I am uncomfortable with how many one-off files though we have for various utilities during training. Can you expose this through the training module `__init__` and make the file "private" (prefix an underscore) so that I can follow up and consolidate some of these utilities?

**Comment by SalmanMohammadi** [comment]
Thanks for adding this @Ankur-singh! Could you also update our recipes to use this, please?
https://github.com/pytorch/torchtune/blob/386ca8d3c543f5a6047699adffae9d10870c2954/recipes/ppo_full_finetune_single_device.py#L571

https://github.com/pytorch/torchtune/blob/386ca8d3c543f5a6047699adffae9d10870c2954/recipes/full_dpo_distributed.py#L584

https://github.com/pytorch/torchtune/blob/386ca8d3c543f5a6047699adffae9d10870c2954/recipes/full_dpo_distributed.py#L497

**Review by joecummings — APPROVED** [review]
🫡
